nautobot_ssot/integrations/servicenow/diffsync/adapter_servicenow.py

```diff
@@ -293,3 +298,23 @@ def sync_complete(self, source, diff, flags=DiffSyncFlags.NONE, logger=None):
         self.bulk_create_interfaces()
 
         source.tag_involved_objects(target=self)
+
+        # If there are objects inside any of the lists in objects_to_delete then iterate over those objects
+        # and remove them from ServiceNow
+        if (
+            self.objects_to_delete["interface"]
+            or self.objects_to_delete["device"]
+            or self.objects_to_delete["product_model"]
+            or self.objects_to_delete["location"]
+            or self.objects_to_delete["company"]
+        ):
+            for grouping in (
+                "interface",
+                "device",
+                "product_model",
+                "location",
+                "company",
+            ):
+                for sn_object in self.objects_to_delete[grouping]:
+                    sn_object.delete()
+                self.objects_to_delete[grouping] = []
```

Comment:
Can we use the `delete` method on the models directly to delete records? I thought this `complete_sync` hook is only necessary so that we can use bulk operations

Reply:
I believe this is done to control the order of objects being deleted to ensure that dependent objects are deleted in right order.

Reply:
I am running the actual delete method from this location in order to control the order in which the delete method is processed for each object. The actual delete method in models.py is not actually deleting the objects but is adding them to "objects_to_delete" dictionary of lists so that they can be processed in the appropriate order (interface > device > product_model > location > company) as that order is different from what is specified in the adapters in "top_level" (company > location). I am using this method because I observed this pattern in the delete method for the ACI integration. 

I will see if I can add some logic to the diffsync/models.py delete method to resolve this.

Reply:
I did a bit of research and attempted to find another way to delete the items in SNOW but was unable to find another method of controller the delete order as the delete order. If this is an issue and you are aware of another method of altering the processing order for deletes please let me know.